public/scripts/project/lock.js

```diff
@@ -0,0 +1,51 @@
+// Our lock system for projects is very simple. If the lock exists,
+// it's almost guaranteed to be valid since the expiry serves
+// to prevent completely locking access to a project in edge cases.
+define(function() {
+  var EXPIRY = 60000; // 1000ms * 60s
```

Comment:
It's a failsafe for the worst possible edge case, where the browser crashes and the lock isn't removed. At least this way the user isn't locked out of their project indefinitely, and doesn't have to open up the dev tools to clear localstorage.